<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <title>Research and Teaching With RevBayes</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="IntroductionToRB_files/reveal.js-3.3.0.1/css/reveal.css"/>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="IntroductionToRB_files/reveal.js-3.3.0.1/css/theme/solarized.css" id="theme">


  <!-- some tweaks to reveal css -->
  <style type="text/css">
    .reveal h1 { font-size: 2.0em; }
    .reveal h2 { font-size: 1.5em;  }
    .reveal h3 { font-size: 1.25em;	}
    .reveal h4 { font-size: 1em;	}

    .reveal .slides>section,
    .reveal .slides>section>section {
      padding: 0px 0px;
    }



    .reveal table {
      border-width: 1px;
      border-spacing: 2px;
      border-style: dotted;
      border-color: gray;
      border-collapse: collapse;
      font-size: 0.7em;
    }

    .reveal table th {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      font-weight: bold;
      border-style: dotted;
      border-color: gray;
    }

    .reveal table td {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      border-style: dotted;
      border-color: gray;
    }


  </style>

    <style type="text/css">code{white-space: pre;}</style>


<!-- Printing and PDF exports -->
<script id="paper-css" type="application/dynamic-css">

/* Default Print Stylesheet Template
   by Rob Glazebrook of CSSnewbie.com
   Last Updated: June 4, 2008

   Feel free (nay, compelled) to edit, append, and
   manipulate this file as you see fit. */


@media print {

	/* SECTION 1: Set default width, margin, float, and
	   background. This prevents elements from extending
	   beyond the edge of the printed page, and prevents
	   unnecessary background images from printing */
	html {
		background: #fff;
		width: auto;
		height: auto;
		overflow: visible;
	}
	body {
		background: #fff;
		font-size: 20pt;
		width: auto;
		height: auto;
		border: 0;
		margin: 0 5%;
		padding: 0;
		overflow: visible;
		float: none !important;
	}

	/* SECTION 2: Remove any elements not needed in print.
	   This would include navigation, ads, sidebars, etc. */
	.nestedarrow,
	.controls,
	.fork-reveal,
	.share-reveal,
	.state-background,
	.reveal .progress,
	.reveal .backgrounds {
		display: none !important;
	}

	/* SECTION 3: Set body font face, size, and color.
	   Consider using a serif font for readability. */
	body, p, td, li, div {
		font-size: 20pt!important;
		font-family: Georgia, "Times New Roman", Times, serif !important;
		color: #000;
	}

	/* SECTION 4: Set heading font face, sizes, and color.
	   Differentiate your headings from your body text.
	   Perhaps use a large sans-serif for distinction. */
	h1,h2,h3,h4,h5,h6 {
		color: #000!important;
		height: auto;
		line-height: normal;
		font-family: Georgia, "Times New Roman", Times, serif !important;
		text-shadow: 0 0 0 #000 !important;
		text-align: left;
		letter-spacing: normal;
	}
	/* Need to reduce the size of the fonts for printing */
	h1 { font-size: 28pt !important;  }
	h2 { font-size: 24pt !important; }
	h3 { font-size: 22pt !important; }
	h4 { font-size: 22pt !important; font-variant: small-caps; }
	h5 { font-size: 21pt !important; }
	h6 { font-size: 20pt !important; font-style: italic; }

	/* SECTION 5: Make hyperlinks more usable.
	   Ensure links are underlined, and consider appending
	   the URL to the end of the link for usability. */
	a:link,
	a:visited {
		color: #000 !important;
		font-weight: bold;
		text-decoration: underline;
	}
	/*
	.reveal a:link:after,
	.reveal a:visited:after {
		content: " (" attr(href) ") ";
		color: #222 !important;
		font-size: 90%;
	}
	*/


	/* SECTION 6: more reveal.js specific additions by @skypanther */
	ul, ol, div, p {
		visibility: visible;
		position: static;
		width: auto;
		height: auto;
		display: block;
		overflow: visible;
		margin: 0;
		text-align: left !important;
	}
	.reveal pre,
	.reveal table {
		margin-left: 0;
		margin-right: 0;
	}
	.reveal pre code {
		padding: 20px;
		border: 1px solid #ddd;
	}
	.reveal blockquote {
		margin: 20px 0;
	}
	.reveal .slides {
		position: static !important;
		width: auto !important;
		height: auto !important;

		left: 0 !important;
		top: 0 !important;
		margin-left: 0 !important;
		margin-top: 0 !important;
		padding: 0 !important;
		zoom: 1 !important;

		overflow: visible !important;
		display: block !important;

		text-align: left !important;
		-webkit-perspective: none;
		   -moz-perspective: none;
		    -ms-perspective: none;
		        perspective: none;

		-webkit-perspective-origin: 50% 50%;
		   -moz-perspective-origin: 50% 50%;
		    -ms-perspective-origin: 50% 50%;
		        perspective-origin: 50% 50%;
	}
	.reveal .slides section {
		visibility: visible !important;
		position: static !important;
		width: auto !important;
		height: auto !important;
		display: block !important;
		overflow: visible !important;

		left: 0 !important;
		top: 0 !important;
		margin-left: 0 !important;
		margin-top: 0 !important;
		padding: 60px 20px !important;
		z-index: auto !important;

		opacity: 1 !important;

		page-break-after: always !important;

		-webkit-transform-style: flat !important;
		   -moz-transform-style: flat !important;
		    -ms-transform-style: flat !important;
		        transform-style: flat !important;

		-webkit-transform: none !important;
		   -moz-transform: none !important;
		    -ms-transform: none !important;
		        transform: none !important;

		-webkit-transition: none !important;
		   -moz-transition: none !important;
		    -ms-transition: none !important;
		        transition: none !important;
	}
	.reveal .slides section.stack {
		padding: 0 !important;
	}
	.reveal section:last-of-type {
		page-break-after: avoid !important;
	}
	.reveal section .fragment {
		opacity: 1 !important;
		visibility: visible !important;

		-webkit-transform: none !important;
		   -moz-transform: none !important;
		    -ms-transform: none !important;
		        transform: none !important;
	}
	.reveal section img {
		display: block;
		margin: 15px 0px;
		background: rgba(255,255,255,1);
		border: 1px solid #666;
		box-shadow: none;
	}

	.reveal section small {
		font-size: 0.8em;
	}

}  
</script>


<script id="pdf-css" type="application/dynamic-css">
    
/**
 * This stylesheet is used to print reveal.js
 * presentations to PDF.
 *
 * https://github.com/hakimel/reveal.js#pdf-export
 */

* {
	-webkit-print-color-adjust: exact;
}

body {
	margin: 0 auto !important;
	border: 0;
	padding: 0;
	float: none !important;
	overflow: visible;
}

html {
	width: 100%;
	height: 100%;
	overflow: visible;
}

/* Remove any elements not needed in print. */
.nestedarrow,
.reveal .controls,
.reveal .progress,
.reveal .playback,
.reveal.overview,
.fork-reveal,
.share-reveal,
.state-background {
	display: none !important;
}

h1, h2, h3, h4, h5, h6 {
	text-shadow: 0 0 0 #000 !important;
}

.reveal pre code {
	overflow: hidden !important;
	font-family: Courier, 'Courier New', monospace !important;
}

ul, ol, div, p {
	visibility: visible;
	position: static;
	width: auto;
	height: auto;
	display: block;
	overflow: visible;
	margin: auto;
}
.reveal {
	width: auto !important;
	height: auto !important;
	overflow: hidden !important;
}
.reveal .slides {
	position: static;
	width: 100%;
	height: auto;

	left: auto;
	top: auto;
	margin: 0 !important;
	padding: 0 !important;

	overflow: visible;
	display: block;

	-webkit-perspective: none;
	   -moz-perspective: none;
	    -ms-perspective: none;
	        perspective: none;

	-webkit-perspective-origin: 50% 50%; /* there isn't a none/auto value but 50-50 is the default */
	   -moz-perspective-origin: 50% 50%;
	    -ms-perspective-origin: 50% 50%;
	        perspective-origin: 50% 50%;
}

.reveal .slides section {
	page-break-after: always !important;

	visibility: visible !important;
	position: relative !important;
	display: block !important;
	position: relative !important;

	margin: 0 !important;
	padding: 0 !important;
	box-sizing: border-box !important;
	min-height: 1px;

	opacity: 1 !important;

	-webkit-transform-style: flat !important;
	   -moz-transform-style: flat !important;
	    -ms-transform-style: flat !important;
	        transform-style: flat !important;

	-webkit-transform: none !important;
	   -moz-transform: none !important;
	    -ms-transform: none !important;
	        transform: none !important;
}

.reveal section.stack {
	margin: 0 !important;
	padding: 0 !important;
	page-break-after: avoid !important;
	height: auto !important;
	min-height: auto !important;
}

.reveal img {
	box-shadow: none;
}

.reveal .roll {
	overflow: visible;
	line-height: 1em;
}

/* Slide backgrounds are placed inside of their slide when exporting to PDF */
.reveal section .slide-background {
	display: block !important;
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	z-index: -1;
}

/* All elements should be above the slide-background */
.reveal section>* {
	position: relative;
	z-index: 1;
}

/* Display slide speaker notes when 'showNotes' is enabled */
.reveal .speaker-notes-pdf {
	display: block;
	width: 100%;
	max-height: none;
	left: auto;
	top: auto;
	z-index: 100;
}

/* Display slide numbers when 'slideNumber' is enabled */
.reveal .slide-number-pdf {
	display: block;
	position: absolute;
	font-size: 14px;
}

</script>


<script>
var style = document.createElement( 'style' );
style.type = 'text/css';
var style_script_id = window.location.search.match( /print-pdf/gi ) ? 'pdf-css' : 'paper-css';
var style_script = document.getElementById(style_script_id).text;
style.innerHTML = style_script;
document.getElementsByTagName('head')[0].appendChild(style);
</script>

</head>
<body>
  <div class="reveal">
    <div class="slides">

<section>
    <h1 class="title">Research and Teaching With RevBayes</h1>
</section>

<section id="overview" class="slide level2">
<h2>Overview</h2>
<ul>
<li>Introduction to graphical modeling</li>
<li>How graphical modeling makes phylogenetic estimation more flexible</li>
<li>Using RevBayes in the classroom</li>
</ul>
</section>
<section id="the-graphical-modeling-framework" class="slide level2">
<h2>The Graphical Modeling Framework</h2>
<ul>
<li>What is a model?</li>
</ul>
</section>
<section id="the-graphical-modeling-framework-1" class="slide level2">
<h2>The Graphical Modeling Framework</h2>
<ul>
<li>What is a model?
<ul>
<li>A model uses mathematics to describe a process, or the behavior of a system</li>
<li>Important facets of the generating process of the data are written down as parameters</li>
</ul></li>
</ul>
</section>
<section id="the-graphical-modeling-framework-2" class="slide level2">
<h2>The Graphical Modeling Framework</h2>
<ul>
<li>What is a model?
<ul>
<li>A model uses mathematics to describe a process, or the behavior of a system</li>
<li>Important facets of the generating process of the data are written down as parameters</li>
</ul></li>
<li>A graphical model represents the relationships and dependencies between model paramters</li>
</ul>
</section>
<section id="a-simple-case" class="slide level2">
<h2>A simple case</h2>
<ul>
<li>An archer: Let’s say the archer is a pretty good archer</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">good_shots &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="dt">n=</span> <span class="dv">100</span>, <span class="dt">mean=</span> <span class="dv">0</span>, <span class="dt">sd =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">hist</span>(good_shots, <span class="dt">main=</span><span class="st">&quot;Archer arrow shots&quot;</span>,  <span class="dt">xlab =</span> <span class="st">&quot;Inches from bullseye&quot;</span>)</a></code></pre></div>
<p><img src="IntroductionToRB_files/figure-revealjs/unnamed-chunk-1-1.png" width="576" /></p>
</section>
<section id="a-simple-case-1" class="slide level2">
<h2>A simple case</h2>
<ul>
<li>An archer: Let’s say the archer is a pretty bad archer</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">bad_shots &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="dt">n=</span> <span class="dv">100</span>, <span class="dt">mean=</span> <span class="dv">0</span>, <span class="dt">sd =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">hist</span>(bad_shots, <span class="dt">main=</span><span class="st">&quot;Archer arrow shots&quot;</span>,  <span class="dt">xlab =</span> <span class="st">&quot;Inches from bullseye&quot;</span>)</a></code></pre></div>
<p><img src="IntroductionToRB_files/figure-revealjs/unnamed-chunk-2-1.png" width="576" /></p>
</section>
<section id="a-simple-case-2" class="slide level2">
<h2>A simple case</h2>
<ul>
<li>It’s easy to describe a model where we are manipulating one parameter</li>
<li>But we rarely have such a simple case when we’re analysing biological data</li>
</ul>
</section>
<section id="a-not-so-simple-case" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<ul>
<li>It’s easy to describe a model where we are manipulating one parameter</li>
<li>But we rarely have such a simple case when we’re analysing biological data</li>
<li>We’re describing the mechanisms that generated our observed data
<ul>
<li>In the case of a tree, this is the process of molecular or morphological evolution</li>
</ul></li>
</ul>
</section>
<section id="a-not-so-simple-case-1" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<ul>
<li>This is where graphical models come in</li>
</ul>
</section>
<section id="a-not-so-simple-case-2" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<ul>
<li>This is where graphical models come in</li>
<li>A <strong>graphical model</strong> represents the joint probability distribution as a graph</li>
</ul>
</section>
<section id="a-not-so-simple-case-3" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li><strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes.</li>
</ul>
</section>
<section id="a-not-so-simple-case-4" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li><strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes.</li>
</ul>
</section>
<section id="a-not-so-simple-case-5" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li><strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes. <img data-src="images/tikz/Normal.tex.png" alt="Image Title" /></li>
</ul>
</section>
<section id="a-not-so-simple-case-6" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li><strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes. <img data-src="images/tikz/Normal.tex.png" alt="Image Title" /></li>
</ul>
<p>Pr( data | model )</p>
</section>
<section id="a-not-so-simple-case-7" class="slide level2">
<h2>A <strong>not so</strong> simple case</h2>
<p><img data-src="images/tikz/graphical_model_legend.png" alt="Image Title" /></p>
</section>
<section id="phylogenetical-complexity" class="slide level2">
<h2>Phylogenetical Complexity</h2>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li><strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes. <img data-src="images/tikz/Mk_model.png" alt="Image Title" /></li>
</ul>
</section>
<section id="phylogenetical-complexity-1" class="slide level2">
<h2>Phylogenetical complexity</h2>
<p><img data-src="images/tikz/constant_nodes.png" alt="Image Title" /></p>
</section>
<section id="phylogenetical-complexity-2" class="slide level2">
<h2>Phylogenetical complexity</h2>
<p><img data-src="images/tikz/stoch_nodes.png" alt="Image Title" /></p>
</section>
<section id="phylogenetical-complexity-3" class="slide level2">
<h2>Phylogenetical complexity</h2>
<p><img data-src="images/tikz/deterministic_node.tex.png" alt="Image Title" /></p>
</section>
<section id="phylogenetical-complexity-4" class="slide level2">
<h2>Phylogenetical complexity</h2>
<p><img data-src="images/tikz/observed_node.tex.png" alt="Image Title" /></p>
</section>
<section id="phylogenetical-complexity-5" class="slide level2">
<h2>Phylogenetical complexity</h2>
<p><img data-src="images/tikz/gtrg_graphical_model.png" alt="Image Title" /></p>
</section>
<section id="who-cares-about-graphical-modeling" class="slide level2">
<h2>Who cares about graphical modeling?</h2>
<p>I’m going to make the case that this framework enables radical transparency in both research and pedagogy.</p>
</section>
<section id="engaging-with-underlying-assumptions" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<ul>
<li>Most phylogenetic estimations treat among-site rate variation as Gamma-distributed</li>
<li>Not every site in a nucleotide alignment evolves at the same rate</li>
<li>Not every character in a morphological dataset evolves at the same rate</li>
</ul>
</section>
<section id="engaging-with-underlying-assumptions-1" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p><img data-src="images/journal.pone.0109210.g003.png" /></p>
</section>
<section id="engaging-with-underlying-assumptions-2" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(dplyr)</a></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">draws =</span> <span class="kw">c</span>(<span class="kw">rgamma</span>(<span class="dt">n =</span> <span class="dv">500</span>, <span class="dt">shape =</span> <span class="fl">.1</span>, <span class="dt">scale =</span> <span class="dv">10</span>),<span class="kw">rgamma</span>(<span class="dt">n =</span> <span class="dv">500</span>, <span class="dt">shape =</span> <span class="fl">.5</span>, <span class="dt">scale =</span> <span class="dv">10</span>)))</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"></a>
<a class="sourceLine" id="cb7-4" data-line-number="4">d2 &lt;-<span class="st"> </span>dat <span class="op">%&gt;%</span><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">lower =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.25</span>), <span class="dt">middle =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.5</span>), <span class="dt">seventyfive =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.75</span>), <span class="dt">upper =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.99</span>))</a>
<a class="sourceLine" id="cb7-5" data-line-number="5"></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x=</span>draws)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> lower)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> middle)) <span class="op">+</span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> seventyfive)) <span class="op">+</span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> upper)) </a></code></pre></div>
<p><img src="IntroductionToRB_files/figure-revealjs/unnamed-chunk-3-1.png" width="576" /></p>
</section>
<section id="engaging-with-underlying-assumptions-3" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p><img data-src="images/lnl.png" alt="Yang1994" /></p>
<p>Yang (1994) demonstrated that modeling ASRV as Gamma distributed with 4 Gamma categories is sufficient</p>
</section>
<section id="engaging-with-underlying-assumptions-4" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p><img data-src="images/Gamma.png" alt="Yang1994" /></p>
<p>Implies many low- or zero-rate characters</p>
</section>
<section id="engaging-with-underlying-assumptions-5" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p><img data-src="images/Fig5.png" alt="Ants5" /></p>
<ul>
<li>We typically don’t collect invarient characters!<br />
</li>
<li>Wagner (2009) and Harisson and Larson (2016) have argued that the Gamma is not appropriate for ASRV in morphological data, and we should use a lognormal</li>
</ul>
</section>
<section id="engaging-with-underlying-assumptions-6" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">draws =</span> <span class="kw">c</span>(<span class="kw">rlnorm</span>(<span class="dt">n =</span> <span class="dv">100</span>, <span class="dt">mean=</span><span class="dv">1</span>)))</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"></a>
<a class="sourceLine" id="cb8-5" data-line-number="5">d2 &lt;-<span class="st"> </span>dat <span class="op">%&gt;%</span><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">lower =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.25</span>), <span class="dt">middle =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.5</span>), <span class="dt">seventyfive =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.75</span>), <span class="dt">upper =</span> <span class="kw">quantile</span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.99</span>))</a>
<a class="sourceLine" id="cb8-6" data-line-number="6"></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x=</span>draws)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> lower)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> middle)) <span class="op">+</span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> seventyfive)) <span class="op">+</span><span class="kw">geom_vline</span>(<span class="dt">data =</span> d2, <span class="kw">aes</span>(<span class="dt">xintercept =</span> upper)) </a></code></pre></div>
<p><img src="IntroductionToRB_files/figure-revealjs/unnamed-chunk-4-1.png" width="576" /></p>
</section>
<section id="engaging-with-underlying-assumptions-7" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<ul>
<li>In previous iterations of software, you might have to email a developer and ask them to implement something</li>
<li>RevBayes’ graphical model framework emphasizes flexibility. We can simply plug in another ASRV distribution.</li>
</ul>
</section>
<section id="engaging-with-underlying-assumptions-8" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p>Open RB_Discrete_Morphology_Model_test.md</p>
</section>
<section id="engaging-with-underlying-assumptions-9" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p>RevBayes can’t be run from within RStudio. In our console, type:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">library</span>(knitr)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw">purl</span>(<span class="st">&quot;RB_MCMC_Discrete_Morph.Rmd&quot;</span>)</a></code></pre></div>
<p>This will generate a RB_MCMC_Discrete_Morph.R file with our Rev code.</p>
</section>
<section id="engaging-with-underlying-assumptions-10" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p>Move the RB_MCMC_Discrete_Morph.R file to scripts, and execute like so:</p>
<pre><code>rb scripts/RB_MCMC_Discrete_Morph.R</code></pre>
<p>This will take about 7 minutes.</p>
</section>
<section id="engaging-with-underlying-assumptions-11" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<p>Return to RB_MCMC_Discrete_Morph.md and scroll to Lognormally-distributed among-character rate variation.</p>
<p>Follow the instructions in the file.</p>
</section>
<section id="engaging-with-underlying-assumptions-12" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
</section>
<section id="engaging-with-underlying-assumptions-13" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
</section>
<section id="engaging-with-underlying-assumptions-14" class="slide level2">
<h2>Engaging with underlying assumptions</h2>
<ul>
<li>Maybe there is an effect here</li>
<li>Dan will show us some better ways of looking at posterior samples</li>
<li>But, the larger point is that <em>flexibility enabled by modularity means that we were able to immediately implement a model we were interested in</em></li>
</ul>
</section>
<section id="transparency-and-documentation" class="slide level2">
<h2>Transparency and documentation</h2>
<p>It wouldn’t be unfair to say there are some things about RevBayes that are more complex than other phylogenetic softwares</p>
</section>
<section id="transparency-and-documentation-1" class="slide level2">
<h2>Transparency and documentation</h2>
<p><img data-src="images/RB_tutorial.png" alt="tutorials" /></p>
</section>
<section id="transparency-and-documentation-2" class="slide level2">
<h2>Transparency and documentation</h2>
<p><img data-src="images/RB_tutorial.png" alt="tutorials" /></p>
<p>Each one is available as a markdown file on the <a href="https://github.com/revbayes/revbayes.github.io">website repository</a></p>
</section>
<section id="transparency-and-documentation-3" class="slide level2">
<h2>Transparency and documentation</h2>
<p>You see an analysis you like? Clone it!</p>
</section>
<section id="transparency-and-documentation-4" class="slide level2">
<h2>Transparency and documentation</h2>
<p>Integrate with various other R packages, like Bookdown, to generate a flexible living textbook for your systematics class!</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">bookdown<span class="op">::</span><span class="kw">render_book</span>(<span class="st">&quot;RB_MCMC_Discrete_Morph.Rmd&quot;</span>, <span class="st">&quot;bookdown::gitbook&quot;</span>)</a></code></pre></div>
</section>
<section id="transparency-and-documentation-5" class="slide level2">
<h2>Transparency and documentation</h2>
<p>We also have the RevKernel, and a set of RevNotebooks ready to go for the Pythonistas.</p>
</section>
    </div>
  </div>

  <script src="IntroductionToRB_files/reveal.js-3.3.0.1/lib/js/head.min.js"></script>
  <script src="IntroductionToRB_files/reveal.js-3.3.0.1/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display the page number of the current slide
        slideNumber: true,
        // Push each slide change to the browser history
        history: true,
        // Vertical centering of slides
        center: false,
        // Opens links in an iframe preview overlay
        previewLinks: true,
        // Transition style
        transition: 'fade', // none/fade/slide/convex/concave/zoom
        // Transition style for full page slide backgrounds
        backgroundTransition: 'default', // none/fade/slide/convex/concave/zoom



        // Optional reveal.js plugins
        dependencies: [
        ]
      });
    </script>
  <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
      var script = document.createElement("script");
      script.type = "text/javascript";
      script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
      document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>

<script>
  (function() {
    if (window.jQuery) {
      Reveal.addEventListener( 'slidechanged', function(event) {  
        window.jQuery(event.previousSlide).trigger('hidden');
        window.jQuery(event.currentSlide).trigger('shown');
      });
    }
  })();
</script>


  </body>
</html>
